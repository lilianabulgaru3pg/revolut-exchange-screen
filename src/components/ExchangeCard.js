import React from 'react';
import { Grid } from '@material-ui/core';
import PropTypes from 'prop-types';
import SelectCurrency from './SelectCurrency';
import CurrentBalance from './CurrentBalance';
import InputAmount from './InputAmount';

import { inputFromType } from '../constants/CommonConstants';

const ExchangeCard = ({
  balance,
  inputVal,
  currency,
  onCurrencyChange,
  onInputChange,
  id,
  currencies,
  type,
  err,
}) => (
  <Grid container>
    <Grid item sm={6} xs={12}>
      <Grid container direction="column">
        <Grid item>
          <SelectCurrency
            id={id}
            currency={currency}
            currencies={currencies}
            onCurrencyChange={onCurrencyChange}
          />
        </Grid>
        <Grid item>
          <CurrentBalance balance={balance} err={err} />
        </Grid>
      </Grid>
    </Grid>
    <Grid item sm={6} xs={12}>
      <InputAmount
        inputVal={inputVal}
        onInputChange={onInputChange}
        autoFocus={type === inputFromType}
      />
    </Grid>
  </Grid>
);

ExchangeCard.propTypes = {
  onCurrencyChange: PropTypes.func.isRequired,
  onInputChange: PropTypes.func.isRequired,
  balance: PropTypes.string.isRequired,
  inputVal: PropTypes.number.isRequired,
  currency: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  err: PropTypes.bool.isRequired,
  currencies: PropTypes.objectOf(PropTypes.string).isRequired,
};
export default ExchangeCard;
