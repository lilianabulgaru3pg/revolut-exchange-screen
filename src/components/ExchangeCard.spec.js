import React from 'react';
import renderer from 'react-test-renderer';
import ExchangeCard from './ExchangeCard';
import {
  CURRENCIES,
  inputFromType,
} from '../constants/CommonConstants';

describe('ExchangeCard Component', () => {
  test('snapshot renders', () => {
    const component = renderer.create(
      <ExchangeCard
        balance=" 11.00 Â£"
        inputVal={240}
        id="select-base-currency"
        currency={CURRENCIES.EUR}
        currencies={CURRENCIES}
        onCurrencyChange={jest.fn()}
        onInputChange={jest.fn()}
        type={inputFromType}
        err
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
