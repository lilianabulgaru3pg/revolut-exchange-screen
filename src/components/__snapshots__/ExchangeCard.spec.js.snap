// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeCard Component snapshot renders 1`] = `
<div
  className="MuiGrid-root MuiGrid-container"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiGrid-root MuiGrid-container"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12"
          >
            <div
              className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-104"
            >
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <div
                  aria-haspopup="listbox"
                  aria-labelledby=" select-base-currency"
                  className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                  id="select-base-currency"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  EUR
                </div>
                <input
                  autoFocus={false}
                  onAnimationStart={[Function]}
                  type="hidden"
                  value="EUR"
                />
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSelect-icon"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorError"
        >
          Balance:
           11.00 Â£
        </h6>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
  >
    <div
      className="MuiTypography-root MuiTypography-body1"
    >
      <input
        autoCapitalize="off"
        autoComplete="off"
        autoCorrect="off"
        autoFocus={true}
        className="sc-AykKC exZWIY"
        onChange={[Function]}
        pattern="[0-9.]*"
        placeholder="0"
        type="number"
        value={240}
      />
    </div>
  </div>
</div>
`;
